src/node/blockstorage.h

```diff
@@ -90,9 +90,9 @@ class BlockManager
      */
     bool LoadBlockIndex()
         EXCLUSIVE_LOCKS_REQUIRED(cs_main);
-    void FlushBlockFile(bool fFinalize = false, bool finalize_undo = false);
-    void FlushUndoFile(int block_file, bool finalize = false);
-    bool FindBlockPos(FlatFilePos& pos, unsigned int nAddSize, unsigned int nHeight, CChain& active_chain, uint64_t nTime, bool fKnown);
+    [[nodiscard]] bool FlushBlockFile(bool fFinalize = false, bool finalize_undo = false);
+    [[nodiscard]] bool FlushUndoFile(int block_file, bool finalize = false);
+    [[nodiscard]] bool FindBlockPos(FlatFilePos& pos, unsigned int nAddSize, unsigned int nHeight, CChain& active_chain, uint64_t nTime, bool fKnown);
     bool FindUndoPos(BlockValidationState& state, int nFile, FlatFilePos& pos, unsigned int nAddSize);
```

Comment:
I can't think of a good reason why this shouldn't be `[[nodiscard]]` too. However, as I said [here](https://github.com/bitcoin/bitcoin/pull/27866#discussion_r1246307365) I am trying to keep this PR's scope limited to just the flush functions and their immediate call sites. I added the `[[nodiscard]]` to `FindBlockPos`, because it reports on the flush error now too. `FindUndoPos` on the other hand is not immediately affected by the flush errors. There are many such examples where we ignore error codes upon return throughout the kernel code and I'd like to tackle them bit by bit.